inmem.go

```diff
@@ -0,0 +1,260 @@
+package clover
+
+import (
+	"sort"
+	"sync"
+)
+
+//iterating over a map doesn't guarantee order of the elements
+//in order to return same results for every unsorted query
+//we also keep insertion id and sort the map elements before querying
+type document struct {
+	insertId int
+	doc      *Document
+}
+
+type inMemEngine struct {
+	sync.Mutex
+	collections map[string]map[string]*document
```

Comment:
It would be better to wrap the value type map[string]*document as a private type collection